src/reactA11yInputElementsRule.ts

```diff
@@ -45,17 +46,34 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 }
 
+function isExcludedInputType(node: ts.JsxSelfClosingElement): boolean {
+    let isExcludedType = false;
+    node.attributes.properties.forEach(
+        (attribute: ts.JsxAttributeLike): void => {
+            if (attribute.kind === ts.SyntaxKind.JsxAttribute) {
```

Comment:
Use `tsutils.isJsxAttribute` on `attribute` and `tsutils.isStringLiteral` on `attribute.initializer. They return whether a given node is of those types, so you won't have to `<>` cast them later on. Nifty hints to the type system.

Reply:
Ah. Thank you! I was digging in the TS source code for these types and couldn't find this one.